### DISCUSSION TITLE
Route functions and client routing

### DISCUSSION THREAD

**Discussion by nadanke** [vikejs/vike#626] [Discussion]
Hey, I'm trying out https://vite-plugin-ssr.com/page-redirection#authentication with client routing. When logging in, the server creates a session and passes the `user` prop down with `pageContext`. When navigating, the part of the page that's reloading gets routed to the login page again - the `login.page.route.ts` file has the `pageRequest.user` prop available server side, but it also runs client side where it's `undefined`, so it routes me back to the login. What's the correct way to handle this? The route function seems to run before everything else, so I don't know how and when to pass down the prop to the client, or why it's not already being passed down since the server passes it with `pageContextInitial` on the navigation request as well.

Any help is greatly appreciated

**Comment by brillout** [comment]
Seems like you didn't add it to https://vite-plugin-ssr.com/passToClient?
